src/js/live-tracker.js

```diff
@@ -0,0 +1,204 @@
+import Component from './component.js';
+import mergeOptions from './utils/merge-options.js';
+
+/* track when we are at the live edge, and other helpers for live playback */
+class LiveTracker extends Component {
+
+  constructor(player, options) {
+    // LiveTracker does not need an element
+    const options_ = mergeOptions({createEl: false}, options);
+
+    super(player, options_);
+
+    this.on(this.player_, 'durationchange', this.handleDurationchange);
+  }
+
+  isBehind() {
+    const liveCurrentTime = this.liveCurrentTime();
+    const currentTime = this.player_.currentTime();
+    const segmentLength = this.segmentLength_ || 12;
+
+    // the live edge window is the amount of seconds away from live
+    // that a player can be, but still be considered live.
+    // we add 0.07 because the live tracking happens every 30ms
+    // and we want some wiggle room for short segment live playback
+    const liveEdgeWindow = segmentLength + 0.07;
+
+    return liveCurrentTime !== Infinity && (liveCurrentTime - liveEdgeWindow) >= currentTime;
+  }
+
+  // all the functionality for tracking when seek end changes
+  // and for tracking how far past seek end we should be
+  trackLive_() {
+    this.pastSeekEnd_ = this.pastSeekEnd_ || 0;
+    const seekable = this.player_.seekable();
+
+    // skip undefined seekable
+    if (!seekable || !seekable.length) {
+      return;
+    }
+
+    const newSeekEnd = this.seekableEnd(0);
```

Comment:
even though I haven't really seen it, might be worth supporting multiple seekable ranges from the beginning

Reply:
we're unsure how that's supposed to work. Any input would be greatly appreciated.